src/main/java/reborncore/common/powerSystem/forge/ForgePowerHandler.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2018 modmuss50 and Gigabit101
+ *
+ *
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ *
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ *
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package reborncore.common.powerSystem.forge;
+
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumParticleTypes;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraft.world.WorldServer;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.energy.CapabilityEnergy;
+import net.minecraftforge.energy.IEnergyStorage;
+import reborncore.api.power.ExternalPowerHandler;
+import reborncore.api.power.IEnergyInterfaceTile;
+import reborncore.common.RebornCoreConfig;
+import reborncore.common.powerSystem.ExternalPowerSystems;
+import reborncore.common.powerSystem.TilePowerAcceptor;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ForgePowerHandler implements ExternalPowerHandler {
+	TilePowerAcceptor powerAcceptor;
+	ForgePowerManager powerManager;
+
+	public ForgePowerHandler(TilePowerAcceptor powerAcceptor) {
+		this.powerAcceptor = powerAcceptor;
+		this.powerManager = new ForgePowerManager(powerAcceptor, null);
+	}
+
+	public void tick() {
+		Map<EnumFacing, TileEntity> acceptors = new HashMap<EnumFacing, TileEntity>();
+		if (powerAcceptor.getEnergy() > 0) { // Tesla or IC2 should handle this if enabled, so only do this without tesla
+			for (EnumFacing side : EnumFacing.values()) {
+				if (powerAcceptor.canProvideEnergy(side)) {
+					TileEntity tile = powerAcceptor.getWorld().getTileEntity(powerAcceptor.getPos().offset(side));
+					if (tile == null) {
+						continue;
+					} else if (ExternalPowerSystems.isPoweredTile(tile)) {
+						// NB: This means we can't register ForgePowerHandler in the ExternalPowerSystems list
```

Comment:
Ah just seen this, working around this might be a better idea. When I have more time Ill think of a better way to do it